packages/astro/components/Code.astro

```diff
@@ -49,6 +50,9 @@ interface Props {
 	 * @default false
 	 */
 	inline?: boolean;
+
+	/** Additional props to pass to the pre element. */
+	additionalProps?: HTMLAttributes<any>;
```

Comment:
I think this shouldn't be a property because you're using the spread operator below. Instead, update the interface Props like so:
```diff
- interface Props {
+ interface Props extends HTMLAttributes<'pre'> {
```

Reply:
I added this in although it was causing problems with the lang property. Specifically LanguageRegistration doesn't seem to bring back a string and HTMLAttributes expects a string for lang. For now I've commented that part out.